### ISSUE TITLE
TIG_LS_REMOTE not behaving as documented

### ISSUE THREAD

**Issue by brtwrst** [jonas/tig#853]
**TLDR:**
TIG_LS_REMOTE default got changed from **git ls-remote** to **git show-ref** in 3448d1cd6a3a3bedd0668c6b6b60fd545a3614ed.
Now using **git ls-remote** in the TIG_LS_REMOTE environment variable breaks TIG because **git ls-remote** uses tabs and **git show-ref** uses spaces to separate output and TIG expects spaces.
**/TLDR**

Please see Chapter 3.2 at http://jonas.nitro.dk/tig/manual.html#env-variables
or Environment Variable Section at https://jonas.github.io/tig/doc/tig.1.html

The Manual speaks of an environment variable TIG_LS_REMOTE that can be used to "limit what branches are shown".

I was playing around with this and immediately noticed some strange behaviour:

When setting the **supposed** default:
```bash
export TIG_LS_REMOTE="git ls-remote ."
```
TIG **breaks** and shows no branches at all.

I checked the source code and found this in [refdb.c](https://github.com/jonas/tig/blob/master/src/refdb.c) line 314:
```c
const char *ls_remote_argv[SIZEOF_ARG] = {
		"git", "show-ref", "--head", "--dereference", NULL
	};
```
which apparently got changed in 3448d1cd6a3a3bedd0668c6b6b60fd545a3614ed 
so when i tried:
```bash
export TIG_LS_REMOTE="git show-ref --head --dereference"
```
everything went back to normal.

when checking the outputs of **git show-ref** vs **git ls-remote** the only difference i could find was that **show-ref** uses a SPACE to separate the output and **ls-remote** uses a TAB.

here is **git ls-remote .**
```
dce1882dc0e27685ee82cf0a80c0dea286e87891        HEAD
dce1882dc0e27685ee82cf0a80c0dea286e87891        refs/heads/master
dce1882dc0e27685ee82cf0a80c0dea286e87891        refs/heads/test
dce1882dc0e27685ee82cf0a80c0dea286e87891        refs/remotes/origin/HEAD
dce1882dc0e27685ee82cf0a80c0dea286e87891        refs/remotes/origin/master
```

and here is **git show-ref --head --dereference**
```
dce1882dc0e27685ee82cf0a80c0dea286e87891 HEAD
dce1882dc0e27685ee82cf0a80c0dea286e87891 refs/heads/master
dce1882dc0e27685ee82cf0a80c0dea286e87891 refs/heads/test
dce1882dc0e27685ee82cf0a80c0dea286e87891 refs/remotes/origin/HEAD
dce1882dc0e27685ee82cf0a80c0dea286e87891 refs/remotes/origin/master
```
and sure enough i found this in [refdb.c](https://github.com/jonas/tig/blob/master/src/refdb.c) line 345
```
code = io_run_load(ls_remote_argv, " ", read_ref, &opt);
```
so it seems the separator TIG expects is hard coded as a space.
I tried replacing the tabs with spaces using `TIG_LS_REMOTE=git ls-remote . | tr '\t' ' '` but that does not work.

**Comment by jonas** [comment]
Thanks for the detailed report. I have been considering removing `TIG_LS_REMOTE` or migrating it to an option in `~/.tigrc` so I am curious if you could share your use case?

Regarding the issue, would you mind trying to change the line you mentioned to

```c
code = io_run_load(ls_remote_argv, " \t", read_ref, &opt);
```

So tabs are also a valid separator? If it works please make a I'll request else if you prefer I can make one.

**Comment by Brtwrst** [comment]
Well i just wanted to decrease visual clutter by (for example) not showing origin/HEAD.
Also i have some remotes that i just use for backup purposes and don't want to see all the time.
I'm not a developer by trade and i've been customizing the settings of tig just for the sake of it.
(Like figuring out how to make my sublimetext work with editor-line-number - sublimetext wants file:line not +line file)
Did i mention i use windows :D
I will try to change the line and compile it on my raspberry pi and get back to you after i figured out how to make a pull request.

BTW thanks for a great program.

**Comment by Brtwrst** [comment]
Alright that was fun.
Changing the line does in fact fix this issue. And i did figure out how pull requests work.
Now i just have to figure out how to actually hide the refs i want to hide.

**Comment by Brtwrst** [comment]
Alright so its a valid use case after all.
Now i'm stuck on the fact that it behaves basically like a whitelist.
For my case here it would be a lot better if it would behave like a blacklist but i can see that - even though i have very little understanding of the source code - this would be a lot more than a one line change. If you ever rewrite the feature to use the .tigrc you might want to look into that possibility.

Thanks again
